### ISSUE TITLE
TypeError: Cannot read property 'hasOwnProperty' of undefined     at Object.DynamoUtil.normalizeItem

### ISSUE THREAD

**Issue by HetIsNiels** [awspilot/dynamodb-oop#28]
I get this weird error:
```
../node_modules/aws-sdk/lib/request.js:31
            throw err;
            ^

TypeError: Cannot read property 'hasOwnProperty' of undefined
    at Object.DynamoUtil.normalizeItem (../node_modules/aws-dynamodb/lib/util.js:103:18)
    at Request.<anonymous> (../node_modules/aws-dynamodb/lib/dynamodb.js:477:60)
    at Request.<anonymous> (../node_modules/aws-dynamodb/lib/dynamodb.js:279:20)
    at Response.<anonymous> (../node_modules/aws-dynamodb/lib/dynamodb.js:274:13)
    at Request.<anonymous> (../node_modules/aws-sdk/lib/request.js:355:18)
    at Request.callListeners (../node_modules/aws-sdk/lib/sequential_executor.js:105:20)
    at Request.emit (../node_modules/aws-sdk/lib/sequential_executor.js:77:10)
    at Request.emit (../node_modules/aws-sdk/lib/request.js:668:14)
    at Request.transition (../node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (../node_modules/aws-sdk/lib/state_machine.js:14:12)
```

Code where it happens:
```javascript
let token = 'x';
let amount = 1;
dynamodb.table('project').where('token').eq(token).index('token').update({
	amount: dynamodb.add(amount)
}
```

No idea why this happens, even updating another column gives this error.

**Comment by databank** [comment]
Sorry for the late delay 

You can not do an update on an index, you need to update the table's row instead.
Also where() needs to match your table's partition/sort definition.

```
either
.where('partition').eq(partition_value).where('sort').eq(sort_value) 
or
.where('partition').eq(partition_value)
```
Please  make sure you are matching table's definition and not the index's definition

**Comment by databank** [comment]
released version 0.1.60 that fixes the crash.
Instead, it will return the error inside the callback normally as `callback(err)`
